File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/refcount_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'refcount'


a = Refcount::A3.new;
b1 = Refcount::B.new a;
b2 = Refcount::B.new a;

if a.ref_count() != 3 
  print "This program will crash... now\n"
end

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/refcount_runme.go
package main

import . "./refcount"

// very innocent example

func main() {
	a := NewA3()
	_ = NewB(a)
	b2 := BCreate(a)

	if a.Ref_count() != 3 {
		panic("This program will crash... now")
	}

	rca := b2.Get_rca()
	// _ = BCreate(rca)
	_ = rca

	if a.Ref_count() != 4 {
		panic("This program will crash... now")
	}

	/* Requires smart pointer support.
	v := NewVector_A(2)
	v.Set(0, a)
	v.Set(1, a)

	_ = v.Get(0)
	DeleteVector_A(v)
	*/
}
